tikv-client/src/main/java/com/pingcap/tikv/expression/visitor/RangePartitionLocator.java

```diff
@@ -0,0 +1,183 @@
+    Object data;
+    Row row = context.getRow();
+    TiTableInfo tableInfo = context.getTableInfo();
+    Expression left = node.getLeft();
+    if (left instanceof ColumnRef) {
+      ColumnRef columnRef = (ColumnRef) left;
+      columnRef.resolve(tableInfo);
+      data = row.get(columnRef.getColumnOffset(), columnRef.getDataType());
+    } else if (left instanceof FuncCallExpr) {
+      // TODO: support more function partition
+      FuncCallExpr partitionFuncExpr = (FuncCallExpr) left;
+      if (partitionFuncExpr.getFuncTp() == YEAR) {
+        data = partitionFuncExpr.eval(Constant.create(row.getDate(0), DateType.DATE)).getValue();
+      } else {
+        throw new UnsupportedOperationException("Partition write only support YEAR() function");
+      }
+    } else {
+      throw new UnsupportedOperationException(
+          String.format("Unsupported expr in range partition %s", left));
+    }
+
+    if (!(node.getRight() instanceof Constant)) {
+      throw new UnsupportedOperationException(
+          String.format("Unsupported right node in partition range expressions %s", node));
+    }
+
+    Constant constant = (Constant) node.getRight();
+    // For the range with single quote such as varchar 'AAAAA' or date'1995-01-01',
+    // we should escape single quote to get the real string need to be compared.
+    String boundString = constant.getValue().toString();
+    if (data instanceof String || data instanceof Date || data instanceof Timestamp) {
+      if (boundString.startsWith("'") && boundString.endsWith("'")) {
+        boundString = boundString.substring(1, boundString.length() - 1);
+      }
+    }
+    Operator comparisonType = node.getComparisonType();
+
+    return evaluateComparison(data, boundString, comparisonType);
+  }
+
+  Boolean evaluateComparison(Object data, String boundString, Operator comparisonType) {
+    // MYSQL IntegerType, we can convert to long and then compare.
+    if (data instanceof Number) {
+      long dataLongValue = ((Number) data).longValue();
+      long bound = Long.parseLong(boundString);
+      switch (comparisonType) {
+        case GREATER_EQUAL:
+          return dataLongValue >= bound;
+        case LESS_THAN:
+          return dataLongValue < bound;
+        default:
+          throw new UnsupportedOperationException("Unsupported comparison type: " + comparisonType);
+      }
+    } else if (data instanceof String) {
+      String dataStringValue = (String) data;
+      switch (comparisonType) {
+        case GREATER_EQUAL:
+          return dataStringValue.compareTo(boundString) >= 0;
+        case LESS_THAN:
+          return dataStringValue.compareTo(boundString) < 0;
+        default:
+          throw new UnsupportedOperationException("Unsupported comparison type: " + comparisonType);
+      }
+    } else if (data instanceof byte[]) {
+      String dataStringValue = new String((byte[]) data);
```

Comment:
Converting bytes to string involves encoding. This uses the platform default charset. Do we need to specify charset?

Reply:
@zhangyangyu Agree with you